app/src/main/java/com/sschuraytz/charcoaldrawing/UndoRedo.java

```diff
@@ -30,6 +31,10 @@ protected void onSizeChanged(int width, int height) {
     public void addBitmap() {
         Bitmap currentTop = currentStack.peek();
         Bitmap newBitmap = Bitmap.createBitmap(currentTop, 0, 0, currentTop.getWidth(), currentTop.getHeight());
+        // to avoid java.lang.OutOfMemoryError
+        if (currentStack.size() == MAX_NUM_OF_BITMAPS) {
+            currentStack.remove(1).recycle();
```

Comment:
You brought this up in our meeting but why 1 and not 0 or currentStack.size()-1?

Reply:
I don't want to use currentStack.size() - 1 since that would remove the most recent part of the drawing. 
I wasn't using 0 because we had created a blank bitmap as the bottom of the stack so there would always be a bitmap to display, and I figured I should leave the blank bitmap. But I guess if users want a clear screen, they can use the "clear" command, so I'll switch this to 0.